hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1418,7 +1419,7 @@ private TracingContext getMetricTracingContext() {
   boolean timerOrchestrator(TimerFunctionality timerFunctionality, TimerTask timerTask) {
     switch (timerFunctionality) {
       case RESUME:
-        if (isMetricCollectionStopped.get()) {
+        if (isMetricCollectionEnabled) {
```

Comment:
The if condition for stopped should also be added here

Reply:
The check is already included within the synchronization block, which is why I didn't add it here again.
synchronized (this) {
            if (isMetricCollectionStopped.get()) {
              resumeTimer();
            }
          }

Reply:
@bhattmanish98 The double-check on isMetricCollectionStopped.get() is to avoid unnecessary synchronization and improve performance

Reply:
@anmolanmol1234 @saikatroy038 Added the check on isMetricCollectionStopped.get(). Please check